### DISCUSSION TITLE
Calling stored procedures with user defined table types (sql server)

### DISCUSSION THREAD

**Discussion by amm1992** [prisma/prisma#11365] [Q&A]
Hi, at my company we have quite a large reliance on stored procedures, as we are writing some new applications I am looking at Prisma to hopefully allow us developer to be able to query the database without the time of our database admins to make more stored procedures who are already swamped. 

However we are hoping we can still use the existing stored procedures as some of them are quite complex and would like to query them.

I can see that there is the functionality to call raw sql as documented [here](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) however I am having issues with certain parameter types as one of the types we need is a User-Defined Table and when I use in the example below, I get the error from the sql server 'nvarchar is incompatible with userDefinedTable' (changed to userDefinedTable from the actual name of the User-Defined table type)
```
let example= await prisma.$queryRaw<Example[]>'EXEC [dbo].[p_getExample] 
    @Param1 = ${request.someNumber}, 
    @UserDefinedParam = ${request.objectResemblingTheUserDefinedTable}, 
    @Param2 = ${request.someString}'
```

My question is, is there any way for me to pass the user defined table type to a sql stored proc in any way with Prisma or is this a potential reason Prisma is not for us?

Thanks

**EDIT**

After having a chat with a colleague he pointed out a solution that is working for me in case anyone else has this issue I have put the example below.

Basically in the raw sql you can declare the param as your user defined table, you can also populate this here from your request values (although I have not in the example), after that you pass this into the param of the executed stored proc. 

```
    let example= await prisma.$queryRaw<Example[]>'
    DECLARE @userDefinedParamValue  [dbo].[userDefinedTable]

    EXEC [dbo].[p_getExample] 
    @Param1 = ${request.someNumber}, 
    @UserDefinedParam = @userDefinedParamValue , 
    @Param2 = ${request.someString}'
```

**Comment by tno2007** [comment]
@amm1992 
Amazing solution. Thank you
